### ISSUE TITLE
Design exploration: Update Vet Center designs to include Events

### ISSUE THREAD

**Issue by mmiddaugh** [department-of-veterans-affairs/va.gov-cms#15349]
## Description
For [Vet Centers 2.0](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/17661), explore how to add event functionality to Vet Center pages. 

Outreach Hub and VAMC editors have access to two content types to feature events on their pages: Event, and Events list. Events are a very important way that Vet Centers and Regional benefit offices provide services. We want to enable this functionality for their facility pages but remain blocked by content build time. 

We should be unblocked in the Spring of 2024 by the AP effort or a retool of the approach to events. In the meanwhile, we can prepare for implementation from a design perspective. 

## Design 
[Vet Center Figma file](https://www.figma.com/file/EVd3q06ukAbVS61Q8MwRAT/Vet-Centers?type=design&node-id=1%3A900&mode=design&t=Wqtm0fJVxdIkoObB-1)

## Design considerations
- Consider aligning with event functionality on VAMC pages. ([Example of a page with events](https://www.va.gov/chicago-health-care/))
- Consider adding events to a side nav. This work was started in [#18121](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/18121) and [#17185](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/17195) when planning satellite location IA. 
- ~~Consider adding event filters mentioned on the [research discovery page](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/facilities/vet-centers/initiatives/2024-vet-center-2.0/seconday-research-synthesis/Vet%20Center%20%26%20VBA%20RO%20events.md).~~ **Not in scope** 
- Consider how to show that events occur in a wide range of time zones, including locations in Guam and the Philippines
- Consider how an image could be added to an event listing
- Consider address any accessibility concerns found in https://github.com/department-of-veterans-affairs/va.gov-cms/issues/16346

## Current state of Vet Center events

Example : [Eventbrite for Staten Island Vet Center (click on Past events)](https://www.eventbrite.com/o/staten-island-vet-center-samuel-b-cottes-jr-32102923355) _(some VCs use Eventbrite as a workaround but it isn't sanctioned by RCS)_

Vet Center events may be different than events at other types of VA facilities 
- events might be open to all (public), open only to Vet Center clients, or open only to Vet Center clients who have been referred by a counselor 
- events can be therapeutic (i.e. facilitated or attended by a counselor) or recreational/social
- some events are limited in the number of participants
- some events are session/cohort based - participants join the event on a particular date or wait until the next session begins
  - (i.e. [Guitars 4 Vets](https://guitars4vets.org/) runs 1:1 sessions for 10 weeks)
- some may have sponsor/partner organizations with websites

## More context
<details>
<summary> Existing event functionality </summary>

- The Event content type allows editors to create, cancel, or reschedule events
- Events exist as individual events and are added to an events list page
- Events can be featured on the health care system homepage
- Event details include start and end date/time with optional recurrence as daily, weekly, or monthly with a recurrence end date until a year into the future.
- Event locations can be specified at a VA facility via dropdown, a non-VA location by adding the address, or online 
- Events can be shared to a global VA.gov events page
- Events can include an event image and registration info

</details>

<details>
<summary> Example VAMC and benefit hub event pages </summary>

- [Event listing page for Houston health care](https://www.va.gov/houston-health-care/events/)

- [Legal Assistance for Veterans and Dependents event page](https://www.va.gov/houston-health-care/events/61545)

- [Outreach hub event listing](https://www.va.gov/outreach-and-events/events/)

</details>

<details>
<summary> Example featured event on Houston health care home page </summary> 

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBbWZHIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--c3581202f014fdc431166fe80c142d108164a202/featured%20event.jpeg)

</details>

<details>
<summary> Open questions </summary>

- Will the existing event functionality meet Vet Center needs? 
- Imagine events for all facility types are tagged to make filtering possible, such as event type (in person vs virtual, event topic (PACT Act vs Education vs Pension), or event audience (family member vs Veteran vs public). If this were possible, what types of filters might be needed for VBA and Vet Center events?  
- Will the "In the Spotlight" feature be deprecated once events are enabled? / Replaced by a featured event slot ala VAMCs?

</details>

<details>
<summary> Problems to solve </summary>

The design should consider the following scenarios from a Veteran-facing perspective
- Vet Center pages have no left nav menu. How might we make events discoverable on these pages? 
   - How might we support expansion of these pages (i.e. support for other location types)? 
- Imagine events for all facility types are tagged to make filtering possible. What is the Veteran experience? How might a Veteran or other audience search for the events they are interested in? How might they learn more about a particular event? 
- Events occur in a wide range of  time zones, including locations in Guam and the Philippines
- How might an image be added to an event listing? 
- Address any accessibility concerns found in https://github.com/department-of-veterans-affairs/va.gov-cms/issues/16346

Support for multiple facility photos images is captured in a separate issue

</details>

## Acceptance criteria
- [x] Design mockups exploring different options are documented in Figma
- [ ] Review with UX Lead, Product, Engineering, Content, and Accessibility
- [ ] Design is reviewed with and sent to VA UX Lead for final sign off
- [ ] Follow up tickets are created
- [ ] Add this work to the CAIA intake ticket because it requires CAIA IA review (likely directly with Mikki, sync with Amanda on this)

**Comment by davidmpickett** [comment]
Note from RO sync today - events will need to support International Addresses

**Comment by jilladams** [comment]
From planning: noting we may want a small research ticket, to do some research discovery around this before we dive into design.

**Comment by jilladams** [comment]
From Michelle: let's split VBA & Vet Center events since Vet Center needs a minor redesign. Research discovery can cover both.

**Comment by jilladams** [comment]
Have split the top level epics for Vet Center vs. VBA events: 
* Vet Centers: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/12926
* VBA: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/17665

UX epics: 
* This epic keeps Vet Center events UX
* VBA moved out to here: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/17666

Research discovery ticket stubbed for @thejordanwood to populate, and is framed out to cover both VBA / VCs and is in both UX epics.

**Comment by davidmpickett** [comment]
@thejordanwood to review async

**Comment by mmiddaugh** [comment]
@thejordanwood @aklausmeier @Agile6MSkinner 
A few responses to the questions under the Open questions heading in the description above...

**Will the existing event functionality meet Vet Center needs?**
- RCS has indicated that Vet Centers will adopt events with existing functionality if it gives them access to events significantly faster than waiting on events 2.0

---

**Imagine events for all facility types are tagged to make filtering possible, such as event type (in person vs virtual, event topic (PACT Act vs Education vs Pension), or event audience (family member vs Veteran vs public). If this were possible, what types of filters might be needed for VBA and Vet Center events?**
- This would make a great office hours topic to solicit editor feedback.

---

**Will the "In the Spotlight" feature be deprecated once events are enabled? / Replaced by a featured event slot ala VAMCs?**
- I will confirm during my meeting with Jessica on June 10th but my assumption is that they will still want spotlights for other use cases, such as to announce upcoming services (i.e. - new MST support group starting soon), news stories, or other use cases.

**Comment by thejordanwood** [comment]
I created mockups that show events on main and satellite locations in [the Figma file](https://www.figma.com/design/EVd3q06ukAbVS61Q8MwRAT/Vet-Centers?node-id=21-2&t=3s7ZOWl3sQDy2xXk-1). In the mockups, I've placed events between In the Spotlight and Services sections. We discussed this in UX sync on 6/20 and have some [outstanding questions](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/17669#issuecomment-2181118389) for this work.  

**What we know:** 

- Events is no longer part of the Vet Centers 2.0 build, but should be part of design
- Events should use existing functionality

**Outstanding questions:** 
- Do satellite location pages get their own events? If so, do they get their own listing page?
- Should there be one or multiple events listed on a main/satellite location page? 
- How many events can be previewed on a main/satellite location page? Currently, VAMCs show one event. Is this a featured event or the next upcoming event?

**Comment by thejordanwood** [comment]
**End of sprint update:** 

This is going to roll over into next sprint. 

- We've asked for stakeholders to provide feedback on events for satellite locations and are hoping to hear on that by the end of the week. 
- There's also a question on using the current VAMC event functionality for Vet Centers. [This slack discussion](https://dsva.slack.com/archives/C0FQSS30V/p1719258112920239) indicates that the VAMC event functionality may be changing to no longer allow featuring events. We need to determine what this means for the Vet Center event work.

**Comment by jilladams** [comment]
Still waiting on stakeholder feedback re: whether Satellite locations will have their own events. 

Re: featuring, still an open Q.

**Comment by davidmpickett** [comment]
@jilladams @Agile6MSkinner - I suggest closing this / moving this to admin/workflow column. This is blocked by stakeholder feedback and will just clutter the board and sprints if we leave it sitting here.

**Comment by thejordanwood** [comment]
We decided in UX sync on 7/18 that we're going to move forward with the feedback we received from stakeholders in the initial meeting on 6/24. Outstanding questions can be identified as I work through this. This is ready to work on.

**Comment by jilladams** [comment]
Work in Sprint 9: identify what we still have questions about, in order to push to stakeholders.

**Comment by thejordanwood** [comment]
After refreshing my memory on this, we still have two outstanding questions: 

- Do events need to show on satellite location pages?
- We want to align with event functionality on VAMC pages. [This slack discussion](https://dsva.slack.com/archives/C0FQSS30V/p1719258112920239) indicates that the VAMC event functionality may be changing to no longer allow featuring events. What does this mean for the event functionality on Vet Centers?

The [design in Figma](https://www.figma.com/design/EVd3q06ukAbVS61Q8MwRAT/Vet-Centers?node-id=21-2&t=lzArcSOk8ecPA3A1-1) shows what events could look like on Vet Center main location pages, satellite location pages, and event list pages. I don't believe there is any design work left for this ticket. The design probably won't change much or at all when these questions get answered. 

Next steps suggestions:

- Asking stakeholders if events should be listed on satellite pages. 
- Creating a new ticket for UX to think through the changing event functionality on VAMC/Vet Center pages. 

cc: @jilladams @Agile6MSkinner @davidmpickett

**Comment by jilladams** [comment]
I chatted about Events featuring with Jordan, who has also talked to @aklausmeier  about it, and just showing the next chronological event (not dealing with featuring) seems smartest, if no one has objections. Jordan is gonna notate the Figma file that way for engineering feedback.

And Jordan noted she has already designed events on Satellite pages, we just need RCS to finalize.

**Comment by davidmpickett** [comment]
Just dug around in [Content Build](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/site/layouts/health_care_region_page.drupal.liquid#L238). to confirm that showing the next chronological event is the current back-up behavior on VAMCs if no event is featured.

**Comment by davidmpickett** [comment]
I'd move to close this ticket as it was meant for initial exploration. We can always have a follow up Design Revisions ticket once we have clarity from Product/Stakeholders on open question and this needs to be finalized

**Comment by thejordanwood** [comment]
Michelle has also sent updated mockups to stakeholders along with the question on if events should show on satellite pages ([relevant Slack thread](https://dsva.slack.com/archives/C0FQSS30V/p1722898027383859?thread_ts=1722373776.820719&cid=C0FQSS30V)).  I'm going to go ahead and close this ticket.
